sbc_email/models/import_letters_history.py

```diff
@@ -379,28 +380,114 @@ def _get_smb_connection(self):
             return SMBConnection(
                 SmbConfig.smb_user, SmbConfig.smb_pass, 'openerp', 'nas')
 
-    @api.model
-    def import_web_letter(self):
+    @api.one
+    def import_web_letter(self, child_code, sponsor_ref, language_name,
+                          original_text, template_name, pdf_letter):
         """
         Call when a letter is set on web site:
             - add web letter to an import set with import letter config
               'Web letter'
         """
-        import_web_letter = self.search([
-            ('config_id.name', '=', 'Web Letter'),
-            ('state', '!=', 'done')])
+        child_id = None
+        if child_code:
+            # Retrieve child code and find corresponding id
+            model_child = self.env['compassion.child'].search(
+                [('code', '=', child_code), ('state', '=', 'P')])
+
+            child_id = model_child.id
+            logger.info("child code found {}".format(model_child.code))
+        else:
```

Comment:
Instead of `else`, I would put `if not child_id` to capture as well if child is not found given a wrong child_code.